server/src/main/java/org/apache/druid/server/http/CoordinatorCompactionConfigsResource.java

```diff
@@ -119,7 +119,8 @@ public Response addOrUpdateCompactionConfig(
           .getCompactionConfigs()
           .stream()
           .collect(Collectors.toMap(DataSourceCompactionConfig::getDataSource, Function.identity()));
-      newConfigs.put(newConfig.getDataSource(), newConfig);
+      DataSourceCompactionConfig updatedConfig = DataSourceCompactionConfig.from(newConfig, current.getEngine());
```

Comment:
I don't think we need this new method `DataSourceCompactionConfig.from()`.
Simply call `ClientCompactionRunnerInfo.validateCompactionConfig()` from here and use the validation result to check if an exception needs to be thrown.